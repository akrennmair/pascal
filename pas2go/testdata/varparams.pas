program test;

var b : integer;

procedure x(var a : integer);
begin
	a := 1
end;

begin
	x(b);
	writeln('b = ', b)
end.
