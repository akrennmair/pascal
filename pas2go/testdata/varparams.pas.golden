package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var (
		b int
	)
	_ = b

	var x func(a *int)
	x = func(a *int) {
		(*a) = 1
		return
	}

	x(&b)
	system.Writeln("b = ", b)
}
